playground/terraform/README.md

```diff
@@ -16,74 +16,114 @@
+project_id           = "project_id"          #Enter the project ID - ID of created project
+gke_name             = "playground-backend"  #Set the GKE name - you can choose any name for Google Kubernetes Engine according to the naming policy above
+region               = "us-east1"            #Set the region - preferred region according to your needs
+location             = "us-east1-b"          #Select the location - location should be in region you set before
+state_bucket         = "bucket_name"         #Name of bucket - Google Cloud bucket where BEAM Playground will put temp files, [terraform state file] (https://spacelift.io/blog/terraform-state)
+bucket_examples_name = "bucket_name-example" #Enter an example bucket name - bucket for some build-in examples for BEAM Playground
+```
+* File name - `state.tfbackend`, item inside:
+```
+bucket               = "bucket_name"         #input bucket name - will be used for terraform tfstate file
+```
+Then, let's configure authentication for the Google Cloud Platform:
 
-* Вeploy backend services
-
-```bash
-./gradlew playground:terraform:deployBackend -Pproject_environment="env-name" -Pdocker-tag="tag"
+* The following command allows us to authenticate using JSON key file
+```
+    export GOOGLE_APPLICATION_CREDENTIALS=`your service account JSON key location` (absolute path)
+```
+* Using the following command, we will activate the newly created service account:
+```
+    gcloud auth activate-service-account `full principal service account` --key-file=`your service account JSON key location` (absolute path)
 ```
 
-* Deploy frontend service
+# Infrastructure deployment:
+* To deploy the Infrastructure, use the following command (please be sure that you are in the "beam" folder):
+```
+./gradlew playground:terraform:InitInfrastructure -Pproject_environment="env" (env - folder name which you created for configuration files)
+```
+# Backend deployment:
+Once the script was executed successfully, you will need to authenticate on Docker and Google Kubernetes Engine
+The following command will authorize us in the Docker registry
+```
+ cat `your service account json key locaton` | docker login -u _json_key --password-stdin https://`chosen_region`-docker.pkg.dev
+```
+* The following command will authenticate us in GKE
+```
+gcloud container clusters get-credentials --region `chosen_pg_location` `gke_name` --project `project_id`
+```
+* We need to create database indexes for BEAM playground examples by the following command:
+```
+gcloud app deploy playground/index.yaml --project=`project_id`
+```
+That's all, the configuration of the environment has been completed. For deploying the backend part to the Google cloud Kubernetes engine, please execute the following command (Ensure you are in the "beam" folder):
+```
+./gradlew playground:terraform:gkebackend -Pproject_environment="env" -Pdocker-tag="tag" (env - folder name which you created for configuration files, tag - image tag for backend)
```

Comment:
Getting error:

```
Caused by: java.io.IOException: Cannot run program "helm" (in directory "<path to project>/playground/terraform"): error=2, No such file or directory
	at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
	... 6 more
Caused by: java.io.IOException: error=2, No such file or directory
	... 7 more

```

When running, `helm version`, I see:

```
version.BuildInfo{Version:"v3.9.4", GitCommit:"dbc6d8e20fe1d58d50e6ed30f09a04a77e4c68db", GitTreeState:"clean", GoVersion:"go1.17.13"}
```

Reply:
@damondouglas not clear from error message snippet - was command run from beam folder (per note above)?

Reply:
Hello, Damon
I hope you're switching branches from the master to the current one. Because, yes, till we merge it to the branch, the helm folder will not be there